Hardware/Hardware/CardReader/StripeCardReader/StripeCardReaderService.swift

```diff
@@ -112,13 +112,24 @@ extension StripeCardReaderService: CardReaderService {
                 return promise(.success(()))
             }
 
+            // The completion block for cancel, apparently, is called when
+            // the SDK has not really transitioned to an idle state.
+            // Clients might need to dispatch operations that rely on this completion block
+            // to start a second operation on the card reader.
+            // (for example, starting an operation after discovery has been cancelled)
+            //
             self?.discoveryCancellable?.cancel { [weak self] error in
-                guard let error = error else {
-                    self?.switchStatusToIdle()
-                    return promise(.success(()))
+                // Horrible, terrible workaround.
+                // And yet, it is the classic "dispatch to the next run cycle".
+                DispatchQueue.main.asyncAfter(deadline: .now() + 0.2) {
```

Comment:
Can we maybe confirm with Stripe if there's a better way to do this?
I enabled Stripe's verbose logging and compared it to our calls, and there's already just 120ms between the cancelable's callback, and what might be the actual internal state update. This is in a relatively dedicated testing device with nothing else running, so I wouldn't be surprised if a more real-world device crosses the 200ms boundary. If we can't find a better way, I would give it a bit more to be safe, maybe one second.

```
2021-06-02 10:18:30.603202+0200 WooCommerce[16043:8119290] 💳 cancelDiscovery called
2021-06-02 10:18:30.603648+0200 WooCommerce[16043:8119290] 💳 cancelDiscovery future called
2021-06-02 10:18:30.610444+0200 WooCommerce[16043:8120605] 💳 l app_id=com.automattic.woocommerce command=discoverReaders completed=0 event=cancel logpoint_level=info scope=SCPCancelable sdk_version=1.4.0 time=1622621910605
2021-06-02 10:18:30.614259+0200 WooCommerce[16043:8120605] 💳 l app_id=com.automattic.woocommerce event=stopBTScan logpoint_level=info scope=SCPBbposBluetoothProximityAdapter sdk_version=1.4.0 time=1622621910610
2021-06-02 10:18:30.618669+0200 WooCommerce[16043:8119290] 💳 discoveryCancellable cancel callback
2021-06-02 10:18:30.618951+0200 WooCommerce[16043:8120605] 💳 l app_id=com.automattic.woocommerce command=discoverReaders event=cancelationSucceeded logpoint_level=info scope=SCPCancelable sdk_version=1.4.0 time=1622621910615
2021-06-02 10:18:30.732585+0200 WooCommerce[16043:8120598] 💳 l app_id=com.automattic.woocommerce command_in_progress=none event=state_change logpoint_level=info scope=SCPTerminal sdk_version=1.4.0 time=1622621910726
2021-06-02 10:18:30.736947+0200 WooCommerce[16043:8120598] 💳 e app_id=com.automattic.woocommerce event=discoverReaders logpoint_level=info scope=SCPTerminal sdk_version=1.4.0 time=1622621910733 total_time_ms=6390
2021-06-02 10:18:30.739693+0200 WooCommerce[16043:8120598] 💳 x domain=terminal_request event=discoverreaders outcome=success scope=scpterminal time=1622621910738
2021-06-02 10:18:30.835547+0200 WooCommerce[16043:8119290] 💳 discoveryCancellable cancel callback after 0.2 second
```

Reply:
Bumped the delay to 1 second